core/src/main/scala/cats/mtl/Handle.scala

```diff
@@ -217,6 +218,41 @@ private[mtl] trait HandleInstances extends HandleLowPriorityInstances {
     }
 }
 
-object Handle extends HandleInstances {
+object Handle extends HandleInstances with HandleVariant {
+
   def apply[F[_], E](implicit ev: Handle[F, E]): Handle[F, E] = ev
+
+  def allowF[F[_], E]: AdHocSyntaxTired[F, E] =
+    new AdHocSyntaxTired[F, E]
+
+  final class AdHocSyntaxTired[F[_], E] {
+
+    def apply[A](body: Handle[F, E] => F[A])(implicit F: ApplicativeThrow[F]): Inner[A] =
+      new Inner(body)
+
+    final class Inner[A](body: Handle[F, E] => F[A])(implicit F: ApplicativeThrow[F]) {
+      def rescue(h: E => F[A]): F[A] = {
```

Comment:
Another trick to avoid allocations, although this slightly changes the ergonomics:
```suggestion
    final class Inner[A](private val body: Handle[F, E] => F[A]) extends AnyVal {
      def rescue(h: E => F[A])(implicit F: ApplicativeThrow[F]): F[A] = {
```